From 739eec9c562d43e75ea1ad477cc7095c4696a480 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:01:48 -0400
Subject: [usb] NS_TO_US should round up

Message-id: <1410887463-20674-37-git-send-email-dzickus@redhat.com>
Patchwork-id: 95650
O-Subject: [RHEL7 PATCH 036/591] USB: NS_TO_US should round up
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 63fb3a280061c5a1d9190015e5a074213f9d23c0
Author: Alan Stern <stern@rowland.harvard.edu>
Date:   Fri Oct 11 11:28:02 2013 -0400

    USB: NS_TO_US should round up

    Host controller drivers use the NS_TO_US macro to convert transaction
    times, which are computed in nanoseconds, to microseconds for
    scheduling.  Periodic scheduling requires worst-case estimates, but
    the macro does its conversion using round-to-nearest.  This patch
    changes it to use round-up, giving a correct worst-case value.

    Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/usb/hcd.h b/include/linux/usb/hcd.h
index fc64b68..dbe3cd1 100644
--- a/include/linux/usb/hcd.h
+++ b/include/linux/usb/hcd.h
@@ -564,9 +564,8 @@ extern void usb_ep0_reinit(struct usb_device *);
    * of (7/6 * 8 * bytecount) = 9.33 * bytecount */
   /* bytecount = data payload byte count */
 
-#define NS_TO_US(ns) ((ns + 500L) / 1000L)
-   /* convert & round nanoseconds to microseconds */
-
+#define NS_TO_US(ns) DIV_ROUND_UP(ns, 1000L)
+   /* convert nanoseconds to microseconds, rounding up */
 
 /*
  * Full/low speed bandwidth allocation constants/support.
-- 
1.7.1